src/lib/operators/table_scan/abstract_table_scan_impl.hpp

```diff
@@ -25,6 +27,9 @@ class AbstractTableScanImpl {
 
   virtual std::shared_ptr<RowIDPosList> scan_chunk(ChunkID chunk_id) const = 0;
 
+  mutable std::atomic<size_t> chunk_scans_skipped{0};
+  mutable std::atomic<size_t> chunk_scans_sorted{0};
```

Comment:
Why does this have to be mutable?

Reply:
`../src/lib/operators/table_scan/column_between_table_scan_impl.cpp:52:9: error: cannot increment value of type 'const std::atomic<size_t>' (aka 'const atomic<unsigned long>')`

Reply:
Why is the Impl const though?

Reply:
It isn't. It's the const methods of the impls as far as I understand.

Reply:
I don't feel strongly about this, but I'd rather remove the consts than add mutables. Resolve as you see fit.